sphinx/texinputs/sphinx.sty

```diff
@@ -682,13 +710,28 @@
 % for captions of literal blocks
 % also define `\theH...` macros for hyperref
 \newcounter{literalblock}
+\ifspx@opt@nonumfigreset
+  \def\theliteralblock {\arabic{literalblock}}
+  \def\theHliteralblock{\arabic{literalblock}}
+\else
+\ifx\spx@opt@numfigreset\@empty
 \ltx@ifundefined{c@chapter}
   {\@addtoreset{literalblock}{section}
   \def\theliteralblock {\ifnum\c@section>\z@ \thesection.\fi\arabic{literalblock}}
   \def\theHliteralblock {\theHsection.\arabic{literalblock}}}
   {\@addtoreset{literalblock}{chapter}
   \def\theliteralblock {\ifnum\c@chapter>\z@ \thechapter.\fi\arabic{literalblock}}
   \def\theHliteralblock {\theHchapter.\arabic{literalblock}}}
+\else
+  \@addtoreset{literalblock}{\spx@opt@numfigreset}
```

Comment:
I will add a check here that the sectioning level determined by ``\spx@opt@numfigreset`` actually exists, using

    \ltx@ifundefined{c@\spx@opt@numfigreset}

like the  ``\ltx@ifundefined{c@chapter}`` used in other branch.